src/FwdState.h

```diff
@@ -20,6 +20,7 @@
 #include "fde.h"
 #include "http/StatusCode.h"
 #include "ip/Address.h"
+#include "ip/NfMarkConfig.h"
```

Comment:
If this include can be dropped or moved, please do so. We should not overexpose all FwdState users to these details.

Reply:
aclFindNfMarkConfig returns Ip::NfMarkConfig, that's why the header is needed.

We could add a TODO to refactor it elsewhere

Reply:
We should `#include "ip/forward.h"` instead then, right? If that file does not forward-declare class NfMarkConfig, then that file should be (trivially) fixed.

Reply:
> Since we return it by copy, don’t we need to know the full class layout?

Those who _call_ the function need to know the class declaration, but all others do not. Rule of thumb: Forwarding declarations (i.e. `foo/forward.h` headers) aside, when a header can be successfully compiled without including header X, it should not include header X.

Reply:
I agree with the principle.
A forward declaration will do; I was under the impression that the full representation of the return type had to be known here, today I learn.
Adding relevant forward declarations in ip/forward.h to be cleaner